### PR TITLE
OCPBUGS-6661, OCPBUGS-9464: Move mTLS CRL handling into the router, and fix accidental duplication of CRLs

### PR DESCRIPTION
**Pull request by rfredette**
Fix #930 by pulling cluster-wide proxy configuration from `proxies.config.openshift.io/cluster` rather than the operator's environment

### CODE REVIEW

**Comment by lihongan** [comment]
to refer https://issues.redhat.com/browse/OCPBUGS-6661

**Comment by frobware** [comment]
TestAWSELBConnectionIdleTimeout failed which is bug https://issues.redhat.com/browse/OCPBUGS-13810.

/test e2e-aws-operator

File: pkg/operator/controller/ingress/controller.go [hunk 1]
```diff
@@ -115,7 +115,7 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 	}); err != nil {
 		return nil, err
 	}
-	// add watch for changes in DNS config
+	// Add watch for changes in DNS config.
```
[the remark this answers is not in the document]
Comment (rfredette) [reply pkg/operator/controller/ingress/controller.go:118]: There are conflicts when backporting due to the fact that go.mod and vendor files are changed by this PR, but as far as I can tell none of the comment changes conflict.

**Comment by rfredette** [comment]
Hit `TestAWSELBConnectionIdleTimeout` failure again.
/retest

**Comment by rfredette** [comment]
More `TestAWSELBConnectionIdleTimeout` failure
/retest

**Comment by lihongan** [comment]
built payload with this PR and tested in https proxy cluster, it works well.
```
$ oc get proxy cluster -oyaml | yq .spec
httpProxy: http://proxy-user1:passwd@10.x.x.x:3128
httpsProxy: https://proxy-user1:passwd@10.x.x.x:3130
noProxy: rhos-d.infra.xxx.com
trustedCA:
  name: user-ca-bundle

sh-4.4$ ls /etc/pki/ca-trust/extracted/pem/
tls-ca-bundle.pem


sh-4.4$ ls ../mtls/latest/ -lh
total 75M
-rw-r--r--. 1 1000610000 root 4.5K Jun  1 07:06 ca-bundle.pem
-rw-r--r--. 1 1000610000 root  75M Jun  1 07:06 crls.pem


$ oc get co/ingress
NAME      VERSION                                                AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
ingress   4.14.0-0.test-2023-06-01-015813-ci-ln-sxdb03k-latest   True        False         False      6h5m    

```

**Comment by rfredette** [comment]
The latest force push removes the additional code for adding the trusted-ca bundle to the router pods. From testing, the only connections that should need the cluster-wide proxy are those for downloading CRLs, and the router does not support downloading CRLs via https. As such, additional code adding the trusted-ca bundle is unnecessary and will only serve to complicate backports.

**Comment by rfredette** [comment]
I've pushed the orphaned commit for trusted-ca bundle support to the branch [rfredette/cluster-ingress-operator:router-https-proxy](https://github.com/rfredette/cluster-ingress-operator/tree/router-https-proxy), but at this time there are no plans to try to merge it.

**Comment by rfredette** [comment]
e2e-aws-operator appears to have passed, but must-gather failed?
/retest

**Pull request merged**